addons/product/models/product_pricelist.py

```diff
@@ -48,6 +48,11 @@ def _default_currency_id(self):
         comodel_name='product.pricelist.item',
         inverse_name='pricelist_id',
         string="Pricelist Rules",
+        domain=[
+            '&',
+            '|', ('product_tmpl_id', '=', None), ('product_tmpl_id.active', '=?', True),
```

Comment:
From the docs `=?` returns true for `None` or `False`. Otherwise the operator behaves like `=`. So I guess you don't need this or logic, as your `=?` already checks for `None`